homeassistant/components/palazzetti/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Palazzetti integration."""
+
+from __future__ import annotations
+
+from pypalazzetti.exceptions import CommunicationError, ValidationError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .coordinator import PalazzettiConfigEntry, PalazzettiDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: PalazzettiConfigEntry) -> bool:
+    """Set up Palazzetti from a config entry."""
+
+    coordinator = PalazzettiDataUpdateCoordinator(hass)
+    try:
+        await coordinator.client.connect()
+        await coordinator.client.update_state()
+    except (CommunicationError, ValidationError) as err:
+        raise ConfigEntryNotReady from err
```

Comment:
can be moved to `_async_setup` in the coordinator (which will be called automatically during async_config_entry_first_refresh), making this a bit cleaner